generators/php/sdk/src/endpoint/http/HttpEndpointGenerator.ts

```diff
@@ -0,0 +1,165 @@
+import { php } from "@fern-api/php-codegen";
+import { HttpEndpoint, ServiceId } from "@fern-fern/ir-sdk/api";
+import { SdkGeneratorContext } from "../../SdkGeneratorContext";
+import { getEndpointReturnType } from "../utils/getEndpointReturnType";
+import { AbstractEndpointGenerator } from "../AbstractEndpointGenerator";
+
+export declare namespace EndpointGenerator {
+    export interface Args {
+        /** the reference to the client */
+        clientReference: string;
+        /** the endpoint for the endpoint */
+        endpoint: HttpEndpoint;
+        /** reference to a variable that is the body */
+        bodyReference?: string;
+    }
+}
+
+const RESPONSE_VARIABLE_NAME = "$response";
+const STATUS_CODE_VARIABLE_NAME = "$statusCode";
+
+export class HttpEndpointGenerator extends AbstractEndpointGenerator {
+    public constructor({ context }: { context: SdkGeneratorContext }) {
+        super({ context });
+    }
+
+    public generate({ serviceId, endpoint }: { serviceId: ServiceId; endpoint: HttpEndpoint }): php.Method {
+        const endpointSignatureInfo = this.getEndpointSignatureInfo({ serviceId, endpoint });
+        const parameters = [...endpointSignatureInfo.baseParameters];
```

Comment:
are we ordering by required parameters first then optional?

Reply:
Yes, but only because we list them in order of path parameters -> request type -> request options. Path parameters are inherently required so we can get away with this for now. 

If we end up needing to support optional path parameters here (which is still unclear whether or not this should be a valid use case), we will need to reorder.